fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/connection/FHIRDbHelper.java

```diff
@@ -0,0 +1,101 @@
+     * could inadvertently leak sensitive information (details of the schema, for example)
+     *
+     * @param logger
+     * @param fx
+     * @param cause
+     * @return
+     */
+    public static <XT extends FHIRPersistenceException> XT severe(Logger logger, XT fx, Throwable cause) {
+        logger.log(Level.SEVERE, fx.getMessage(), cause);
+        return fx;
+    }
+    
+    /**
+     * Log the exception message here along with the cause stack. Return the
+     * exception fx to the caller so that it can be thrown easily.
+     *
+     * @param logger
+     * @param fx
+     * @param errorMessage
+     * @param cause
+     * @return
+     */
+    public static <XT extends FHIRPersistenceException> XT severe(Logger logger, XT fx, String errorMessage,
+            Throwable cause) {
+        if (cause != null) {
+            logger.log(Level.SEVERE, fx.addProbeId(errorMessage), cause);
+        } else {
+            logger.log(Level.SEVERE, fx.addProbeId(errorMessage));
+        }
+        return fx;
+    }
+    
+    public static FHIRPersistenceDataAccessException buildExceptionWithIssue(String msg, IssueType issueType)
+            throws FHIRPersistenceDataAccessException {
+        Issue ooi = FHIRUtil.buildOperationOutcomeIssue(msg, issueType);
+        return new FHIRPersistenceDataAccessException(msg).withIssue(ooi);
+    }
+
+    public static FHIRPersistenceDBConnectException buildFHIRPersistenceDBConnectException(String msg, IssueType issueType)
+            throws FHIRPersistenceDBConnectException {
+        Issue ooi = FHIRUtil.buildOperationOutcomeIssue(msg, issueType);
+        return new FHIRPersistenceDBConnectException(msg).withIssue(ooi);
+    }
+
+    /**
+     * Close the connection if not null.
+     * @implNote This connection object is just a wrapper. If we're in a 
+     *           transaction, then closing it doesn't do much, other than tell
+     *           the transaction manager that the connection is no longer in
+     *           use. The transaction manager still holds the underlying 
+     *           database connection open, and will use that connection the 
+     *           next time getConnection() is called for the same datasource 
+     *           within this thread. Only when the transaction commits 
+     *           will the connection be returned to the pool (or closed). 
+     *           If connections remain open when commit() is called, the 
+     *           transaction will fail.
+     * @param c
+     */
+    public static void close(Connection connection) {
+        if (connection != null) {
+            try {
+                connection.close();
+            } catch (Throwable e) {
+                // log the failure, but suppress the exception
+                FHIRPersistenceDBCleanupException ce = new FHIRPersistenceDBCleanupException("Failure closing Connection.", e);
+                log.log(Level.SEVERE, ce.getMessage(), ce);
```

Comment:
you pass in some methods, and not in others. 
I do like the concept of the helper, I just prefer the traceability of a logged message being identified in the originating class.

Reply:
Agreed. There may have been some cases where we didn't have a logger at hand when wanting to make this call. In this case, there's probably plenty of drama logged already before the close is called, so we won't be left wanting for context.

Reply:
removed method